src/app/shared/services/doctors.service.ts

```diff
@@ -1,32 +1,160 @@
-import { Injectable } from '@angular/core';
-import { BehaviorSubject, map, Observable } from 'rxjs';
-import { Doctor } from '../components/doctor-card/doctor.model';
+import { Injectable, OnDestroy } from '@angular/core';
 import { AngularFireDatabase } from '@angular/fire/compat/database';
+import { BehaviorSubject, Subscription } from 'rxjs';
+
+import { Doctor } from '../components/doctor-card/doctor.model';
+import { SortingTypes } from '../models/sorting-types.model';
+import { DoctorsSpecialty } from '../models/doctors-specialty.model';
+import { get } from 'lodash';
+
+const DOCTORS_PER_PAGE = 6;
 
 @Injectable({
   providedIn: 'root',
 })
-export class DoctorsService {
-  doctorsSubject = new BehaviorSubject<Doctor[]>([]);
-  doctors = this.doctorsSubject.asObservable();
+export class DoctorsService implements OnDestroy {
+  private visibleDoctorsSubject = new BehaviorSubject<Doctor[]>([]);
+  private visibleSpecialtiesSubject = new BehaviorSubject<DoctorsSpecialty[]>(
+    []
+  );
+
+  private topDoctorsSubject = new BehaviorSubject<Doctor[]>([]);
+  private paginationSizeSubject = new BehaviorSubject<number>(0);
+
+  private paginationCurrentSubject = new BehaviorSubject<number>(0);
+  private sortingTypeSubject = new BehaviorSubject<string>('');
+  private searchQuerySubject = new BehaviorSubject<string>('');
+
+  private subsription!: Subscription;
+
+  visibleDoctors = this.visibleDoctorsSubject.asObservable();
+  visibleSpecialties = this.visibleSpecialtiesSubject.asObservable();
+
+  paginationCurrent = this.paginationCurrentSubject.asObservable();
+  paginationSize = this.paginationSizeSubject.asObservable();
+  sortingType = this.sortingTypeSubject.asObservable();
+
+  topDoctors = this.topDoctorsSubject.asObservable();
```

Comment:
Top doctors list seem to have little logical connection with filtering/sorting/pagination. I suggest to divide the logic and move topDoctor service in the separate file.